vite.config.js

```diff
@@ -9,6 +9,34 @@ import autoprefixer from 'autoprefixer';
 import fs from 'fs';
 import path from 'path';
 
+const hydeVitePlugin = () => ({
+    name: 'hyde-vite',
+    configureServer(server) {
+        // Create hot file when Vite server starts
+        fs.writeFileSync(path.resolve(process.cwd(), 'app/storage/framework/cache/vite.hot'), '');
+
+        // Remove hot file when Vite server closes
+        ['SIGINT', 'SIGTERM'].forEach(signal => {
+            process.on(signal, () => {
+                fs.rmSync(path.resolve(process.cwd(), 'app/storage/framework/cache/vite.hot'));
+                process.exit();
+            });
+        });
```

Comment:
### Options Tried

1. **Vite Plugin Lifecycle Hooks**:
   - **`closeBundle`**: Intended to clean up after the build process, but it didn't trigger reliably in this case.
   - **`buildEnd`**: Meant to handle cleanup after the build, but similarly, it didn't work as expected.
   - **Server Events**: Attempted to use `server.httpServer.on('close')` and `server.watcher.on('close')`, but these events also failed to trigger consistently during development.

2. **Process Signal Listeners**:
   - **`SIGINT`**: Captured when the user manually stops the server (e.g., using Ctrl+C). This worked reliably and allowed for cleanup of the `vite.hot` file.
   - **`SIGTERM`**: Sent by process managers or during system shutdown. Included for completeness.

### Rationale for Using Node.js Signal Listeners

- **Reliability**: Signal listeners for `SIGINT` and `SIGTERM` provide a dependable way to handle cleanup operations. They ensure that applications can respond to termination requests, allowing for graceful shutdowns.
  
- **Simplicity**: Using process signals simplifies the code by focusing on a straightforward mechanism for cleanup, rather than relying on various Vite lifecycle hooks and server events that may not trigger consistently.

- **Common Practice**: Handling process signals is a standard practice in Node.js applications, especially for server-side code. It ensures that resources are released properly, preventing potential issues like file locks or incomplete operations.

- **Flexibility**: By listening for both `SIGINT` and `SIGTERM`, the application can handle different shutdown scenarios, making it more robust in various environments (development, production, containerized setups).

In conclusion, while various options were explored, the use of Node.js signal listeners simed to be the most effective and reliable method for managing cleanup tasks in the Vite plugin.